python_bindings/dda_model/binarizers.py

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+import math
+import matplotlib.pyplot as plt
+
+def parabolic(parameters):
+    return parameters - np.square(parameters)
+
+def gradParabolic(parameters):
+    return 1 - 2 * parameters
+
+def gaussian(parameters, sig, mu):
+    return 1/(np.sqrt(2*np.pi)*sig)*np.exp(-np.power((parameters - mu)/sig, 2)/2)/24
+
+def gradGaussian(parameters, sig, mu):
+    return -(parameters - mu) * gaussian(parameters, sig, mu) / np.square(sig)
+
+def triangular(parameters, slope):
+    result = np.array(parameters)
+
+    for i, para in enumerate(parameters):
+        if(para <= 0.5):
+            result[i] = slope*para
+        else:
+            result[i] = slope*(1-para)
+
+    return result
+
+def gradTriangular(parameters, slope):
+    result = np.array(parameters)
+
+    for i, para in enumerate(parameters):
+        if(para <= 0.5):
+            result[i] = slope
+        else:
+            result[i] = -slope
+
+    return result
+
+def piecewise_update(x, x_max, y_min, y_max):
```

Comment:
instead of hardcoding at 0.5, you can have the start of ramp-up and end of ramp-up be settable parameters with default values.

Reply:
done!